packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -42,8 +42,11 @@ const signedComponentToMetaMap: Map<ComponentConstructor, ComponentMeta> = new M
  */
 export function registerComponent(
     Ctor: ComponentConstructor,
-    { name, tmpl: template }: { name: string; tmpl: Template }
+    { name, tmpl: template }: { name: string | undefined; tmpl: Template }
 ): ComponentConstructor {
+    if (isUndefined(name)) {
```

Comment:
any particular reason for the compiler to provide an undefined name?

Reply:
It has never been passed as far as I can tell: https://github.com/salesforce/lwc/blob/20275fc44b04e025ae10a2a142bf984c0c0f4f52/packages/%40lwc/babel-plugin-component/src/post-process/transform.js#L117-L120.

At best the compiler will be able to set the `name` as `Ctor.name`. IMO we should remove the optional `name` and just retrieve it from the `Ctor.name`.

Reply:
I agreed with that assessment because the compiler can't take it out of the Ctor, because it might be circular, and who knows what other quirk in platform. I'm ok with the runtime inference here.